### PR TITLE
feat: [CDS-77249]: Added classes for ECR config in OCI Helm Connector

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Added classes for ECR config in OCI Helm Connector
Screenshot 2023-08-22 at 4 12 01 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51899)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigType.java [hunk 1]
```diff
@@ -7,10 +7,15 @@
...
 package io.harness.cdng.manifest.yaml.oci;
 
+import io.harness.annotations.dev.CodePulse;
+import io.harness.annotations.dev.HarnessModuleComponent;
 import io.harness.annotations.dev.HarnessTeam;
 import io.harness.annotations.dev.OwnedBy;
+import io.harness.annotations.dev.ProductModule;
 
 @OwnedBy(HarnessTeam.CDP)
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
```
Comment (abhinavsingh1196) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigType.java:17]: unitCoverageRequired should be false for entities?

**Review by acasian-harness — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger ut4, ut0, ut9

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigType.java (hunk 1 above)
Comment (GoelPratyush99) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigType.java]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+@Builder
+@EqualsAndHashCode(callSuper = false)
+@JsonTypeName(ECR)
+@SimpleVisitorHelper(helperClass = ConnectorRefExtractorHelper.class)
+@TypeAlias("OciEcrConfig")
+@RecasterAlias("io.harness.cdng.manifest.yaml.OciHelmChartStoreEcrConfig")
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false, components = {HarnessModuleComponent.CDS_K8S})
+public class OciHelmChartStoreEcrConfig implements OciHelmChartStoreConfig, Visitable, WithConnectorRef {
+  @Wither @NotNull @YamlSchemaTypes({string}) private ParameterField<String> connectorRef;
+  @Wither @NotNull @ApiModelProperty(dataType = STRING_CLASSPATH) private ParameterField<String> region;
```
Comment (abosii-harness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java:51]: can you add @NotEmpty as well?

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java` line 50 at r2](https://reviewable.io/reviews/harness/harness-core/51899#-NcVkLDGAMcYOYzf4cD5:-NcVkLDGAMcYOYzf4cD6:b133080) ([raw file](https://github.com/harness/harness-core/blob/92ba8df33f357179bfb032e3de1a331099c718dd/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java#L50)):*
> ```Java
> @CodePulse(module = ProductModule.CDS, unitCoverageRequired = false, components = {HarnessModuleComponent.CDS_K8S})
> public class OciHelmChartStoreEcrConfig implements OciHelmChartStoreConfig, Visitable, WithConnectorRef {
>   @Wither @NotNull @YamlSchemaTypes({string}) private ParameterField<String> connectorRef;
> ```

don't we need ApiModelProperty here?

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java` line 51 at r2](https://reviewable.io/reviews/harness/harness-core/51899#-NcVkZnB4HgGO1XECS-w:-NcVkZnB4HgGO1XECS-x:bfa8g72) ([raw file](https://github.com/harness/harness-core/blob/92ba8df33f357179bfb032e3de1a331099c718dd/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java#L51)):*
> ```Java
> public class OciHelmChartStoreEcrConfig implements OciHelmChartStoreConfig, Visitable, WithConnectorRef {
>   @Wither @NotNull @YamlSchemaTypes({string}) private ParameterField<String> connectorRef;
>   @Wither @NotNull @ApiModelProperty(dataType = STRING_CLASSPATH) private ParameterField<String> region;
> ```

don't we need YamlSchemaType here?

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 6 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51899)* status: 8 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by alebuh — APPROVED** [review]

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java (hunk 2 above)
Comment (GoelPratyush99) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreEcrConfig.java]: Done

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 4 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51899)* status: 8 file(s) to review. 2. unresolved discussions. Waiting on team(s): k8s-team.

**Comment by GoelPratyush99** [comment]
trigger review

**Pull request merged**